### ISSUE TITLE
Creating a playlist when music is on external sdcard due to insufficient permissions

### ISSUE THREAD

**Issue by theoklink** [Iamsdt/MusicPlayer#1]
As the title explains, the insertion into a playlist  fails for android 10  and above 
2020-07-15 12:10:05.532 7977-8060/com.iamsdt.musicplayer E/UCEHandler: App crashed, executing UCEHandler's UncaughtExceptionHandler
    java.lang.SecurityException: com.iamsdt.musicplayer has no access to content://media/external_primary/audio/media/19312
        at android.os.Parcel.createException(Parcel.java:2087)
        at android.os.Parcel.readException(Parcel.java:2055)
        at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:188)
        at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:140)
        at android.content.ContentProviderProxy.bulkInsert(ContentProviderNative.java:503)
        at android.content.ContentResolver.bulkInsert(ContentResolver.java:1996)
        at com.example.musicplayer.data.repo.PlaylistRepository.addToPlaylist(PlaylistRepo.kt:103)
        at com.example.musicplayer.ui.songs.SongVM$addToPlaylist$1.invokeSuspend(SongVM.kt:72)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(Dispatched.kt:241)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:594)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.access$runSafely(CoroutineScheduler.kt:60)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:740)
2020-07-15 12:10:05.570 7977-8060/com.iamsdt.musicplayer I/Process: Sending signal. PID: 7977 SIG: 9

**Comment by BilalAsif25** [comment]
same issue, can you please do something about it @Iamsdt

**Comment by BilalAsif25** [comment]
@theoklink have you managed to resolve it bro ?

**Comment by theoklink** [comment]
Android 10 is already a nightmare but android 11 even more so.

**Comment by Iamsdt** [comment]
I agree with you @theoklink. Doc suggests, use this
```
android:requestLegacyExternalStorage="true"
```
But this already added into AndroidManifest.xml

I found a stack overflow question, you can take a look
https://stackoverflow.com/questions/57398381/api-29-mediastore-access

**Comment by theoklink** [comment]
Yes, my post from a while back. Still no further forward

**Comment by theoklink** [comment]
No, still stuck. It is ok when there is no sdcard. Tracks on sdcard have no
permissions

**Comment by Iamsdt** [comment]
Yeah, external storage creates this issue.
If you figure the solution, please open a PR
